contracts/extensions/votingReputation/VotingReputationStorage.sol

```diff
@@ -197,17 +200,24 @@ contract VotingReputationStorage is ColonyExtension, BasicMetaTransaction, Votin
       ) {
         return MotionState.Closed;
       } else {
-        return finalizableOrFinalized(motion.action);
+        return finalizableOrFinalized(_motionId);
       }
     }
   }
 
   // If we decide that the motion is finalizable, we might actually want it to
   //  report as finalized if it's a no-action motion.
-  function finalizableOrFinalized(bytes memory action) internal pure returns (MotionState) {
-    bytes4 sig;
-    assembly { sig := mload(add(action, 0x20)) }
-    return sig == NO_ACTION ? MotionState.Finalized : MotionState.Finalizable;
+  function finalizableOrFinalized(uint256 _motionId) internal view returns (MotionState) {
+    Motion storage motion = motions[_motionId];
+    if (motion.sig == NO_ACTION || getSig(motion.action) == NO_ACTION) {
+      return MotionState.Finalized;
+    } else if (motion.sig == bytes4(0x0) && getSig(motion.action) == MULTICALL) {
+      // (Inefficiently) handle the potential case of a v9 motion
+      ActionSummary memory actionSummary = getActionSummary(motion.action, getTarget(motion.altTarget));
```

Comment:
I actually don't think that would be correct behavior here. If someone creates an "invalid" multicall in version 9, and then upgrades, the motion would be stuck in perpetuity (since the downstream functions would always error).

I think our options are:

- Allow invalid multicalls created before the upgrade to be executed
- Prevent invalid multicalls from being executed by returning "finalized" instead of "finalizable"

Reply:
I've implemented this to tag any invalid summaries which would otherwise be "finalizable" as "finalized".

Reply:
In which case, the comments in `getActionSummary` need updating.

Reply:
I'm talking about the comments in `getActionSummary` that say `Invalid multicall, caller should error`.